api/envoy/config/route/v3/route_components.proto

```diff
@@ -457,6 +457,28 @@ message RouteMatch {
     // Note that CONNECT support is currently considered alpha in Envoy.
     // [#comment:TODO(htuch): Replace the above comment with an alpha tag.
     ConnectMatcher connect_matcher = 12;
+
+    // Supports segment based url_template matching for *:path" header.
+    // This match is always case sensitive.
+    // Its syntax is defined at `http.proto <https://github.com/googleapis/googleapis/blob/f829b1334cce86aa3738f3c0698d814b56664445/google/api/http.proto#L208>`_.
+    //
+    // Syntax highlights:
+    //
+    // .. code-block:: text
+    //
+    // * wildcard "*": matchs only 1 path segment.
+    // * double wildcard "**": matches 0 or more path segments.
+    // * variable /foo/{x=**}: group path segments to a variable.
+    // * variable /foo/{x}: same as /foo/{x=*}
+    //
+    // For examples:
+    //
+    // .. code-block:: text
+    //
+    // * /foo/*: "/foo/bar" should match. But "/foo" and "/foo/bar1/bar2" should not match.
+    // * /foo/**: "/foo", "/foo/bar", "/foo/bar1/bar2" all should match.
+    //
+    string url_template = 13;
```

Comment:
Arguably now might be the time to put this in `StringMatcher` and make `StringMatcher` the new way and deprecate existing matchers. Not sure if it has to be in this PR, but it feels like straw-breaking-camel-back.

Reply:
Prefer to do that in separate PR.

Reply:
That is exactly in the comment of safe_regex

>     // [#next-major-version: In the v3 API we should redo how path specification works such
    // that we utilize StringMatcher, and additionally have consistent options around whether we
    // strip query strings, do a case sensitive match, etc. In the interim it will be too disruptive
    // to deprecate the existing options. We should even consider whether we want to do away with
    // path_specifier entirely and just rely on a set of header matchers which can already match
    // on :path, etc. The issue with that is it is unclear how to generically deal with query string
    // stripping. This needs more thought.]

Since it will be a large one anyway, let's keep this PR simple.

Reply:
Fair.